src/seqme/models/hyformer.py

```diff
@@ -0,0 +1,191 @@
+from enum import Enum
+
+import numpy as np
+import torch
+from tqdm import tqdm
+
+from .exceptions import OptionalDependencyError
+
+
+class HyformerCheckpoint(str, Enum):
+    # molecules checkpoints
+    molecules_8M = "SzczurekLab/hyformer_molecules_8M"
+    molecules_50M = "SzczurekLab/hyformer_molecules_50M"
+
+    # peptides checkpoints
+    peptides = "SzczurekLab/hyformer_peptides"
+    peptides_mic = "SzczurekLab/hyformer_peptides_mic"
+
+
+class Hyformer:
+    """
+    Wrapper for the Hyformer molecule/peptide embedding model.
+
+    Computes sequence-level embeddings by extracting the [CLS] token embedding.
+
+    Installation: ``pip install seqme[hyformer]``
+
+    Reference:
+        Izdebski et al., "Synergistic Benefits of Joint Molecule Generation and Property Prediction"
+        (https://arxiv.org/abs/2504.16559)
+    """
+
+    def __init__(
+        self,
+        model_name: HyformerCheckpoint | str,
+        *,
+        device: str | None = None,
+        batch_size: int = 256,
+        verbose: bool = False,
+    ):
+        """
+        Initialize Hyformer model.
+
+        Args:
+            model_name: Model checkpoint name or enum.
+            device: Device to run inference on, e.g., "cuda" or "cpu".
+            batch_size: Number of sequences to process per batch.
+            verbose: Whether to display a progress bar.
+        """
+        if isinstance(model_name, HyformerCheckpoint):
+            model_name = model_name.value
+
+        if device is None:
+            device = "cuda" if torch.cuda.is_available() else "cpu"
+
+        self.batch_size = batch_size
+        self.device = device
+        self.verbose = verbose
+
+        try:
+            from hyformer import AutoModel, AutoTokenizer
+            from hyformer.utils import create_dataloader
+        except ModuleNotFoundError:
+            raise OptionalDependencyError("hyformer") from None
+
+        self.tokenizer = AutoTokenizer.from_pretrained(model_name)
+        self.model = AutoModel.from_pretrained(model_name)
+        self._create_dataloader_fn = create_dataloader
+
+        self.model.to(device)
+        self.model.eval()
+
+    def __call__(self, sequences: list[str]) -> np.ndarray:
+        return self.embed(sequences)
+
+    def generate(self, sequences: list[str]) -> np.ndarray:
+        pass
+
+    def embed(self, sequences: list[str]) -> np.ndarray:
+        """
+        Compute embeddings for a list of sequences.
+
+        Each sequence is tokenized and passed through the model.
+        Token embeddings are averaged (excluding special tokens) to produce a single embedding per sequence.
```

Comment:
Sequence embeddings use the CLS tokens. Please update this.

Reply:
done